web/src/main/java/com/epam/esm/web/controller/CertificateController.java

```diff
@@ -0,0 +1,99 @@
+package com.epam.esm.web.controller;
+
+import com.epam.esm.domain.dto.CertificateDTO;
+import com.epam.esm.service.CertificateService;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+/**
+ * The class {@code CertificateController} handles requests to perform certificate operations
+ *
+ * @author Maksim Zarutski
+ */
+@RestController
+@RequestMapping("/api")
+public class CertificateController {
+
+    private final CertificateService certificateService;
+
+    public CertificateController(CertificateService certificateService) {
+        this.certificateService = certificateService;
+    }
+
+    /**
+     * Gets all certificates with tags matching search parameters in defined order.
+     * Search parameters are optional
+     *
+     * @param tagName         tag name
+     * @param certificateName part of the certificate's name
+     * @param description     part of the certificate's description
+     * @param sortBy          sort type
+     * @param order           order type
+     * @return List of certificates found
+     */
+    @GetMapping("/gift_certificates/search")

```

Comment:
You do not have to provide "gift_certificates" url part for every request.

Reply:
thanks. good point. fixed